File: backend/app/schemas/intent.py
from enum import Enum
from typing import List, Dict, Any, Optional
from pydantic import BaseModel, Field, ConfigDict

class IntentType(str, Enum):
    LEARN = "learn"       # 学习新领域 (生成知识图谱)
    RESEARCH = "research" # 深入研究 (查找深度内容)
    TRACK = "track"       # 追踪动态 (订阅更新)
    CREATE = "create"     # 手动创建 (指定结构)
    GENERAL = "general"   # 通用/其他

class ActionType(str, Enum):
    CREATE_NODE = "create_node"
    CREATE_CLUSTER = "create_cluster"
    SEARCH_CONTENT = "search_content"
    GENERATE_REPORT = "generate_report"
    SUBSCRIBE_TOPIC = "subscribe_topic"

class Action(BaseModel):
    type: ActionType
    description: str
    parameters: Dict[str, Any]
    priority: int = 1
    
    model_config = ConfigDict(from_attributes=True)

class IntentParseResult(BaseModel):
    original_text: str
    intent_type: IntentType
    primary_topic: str
    sub_topics: List[str] = []
    goal: Optional[str] = None
    parameters: Dict[str, Any] = {}
    confidence: float = 1.0
    
    model_config = ConfigDict(from_attributes=True)

class IntentCreateRequest(BaseModel):
    text: str = Field(..., min_length=2, max_length=500)
    source: str = "user_input" # user_input, voice, etc.

class IntentResponse(BaseModel):
    id: str # Temporary ID or DB ID
    parsed_intent: IntentParseResult
    suggested_actions: List[Action] = []
    
    model_config = ConfigDict(from_attributes=True)

File: frontend/src/lib/api/intent.ts
import api from '@/lib/api';

export enum IntentType {
  LEARN = "learn",
  RESEARCH = "research",
  TRACK = "track",
  CREATE = "create",
  GENERAL = "general"
}

export enum ActionType {
  CREATE_NODE = "create_node",
  CREATE_CLUSTER = "create_cluster",
  SEARCH_CONTENT = "search_content",
  GENERATE_REPORT = "generate_report",
  SUBSCRIBE_TOPIC = "subscribe_topic"
}

export interface Action {
  type: ActionType;
  description: string;
  parameters: Record<string, any>;
  priority: number;
}

export interface IntentParseResult {
  original_text: string;
  intent_type: IntentType;
  primary_topic: string;
  sub_topics: string[];
  goal?: string;
  parameters: Record<string, any>;
  confidence: number;
}

export interface IntentResponse {
  id: string;
  parsed_intent: IntentParseResult;
  suggested_actions: Action[];
}

export const intentApi = {
  parse: async (text: string) => {
    const response = await api.post('/intents/parse', { text });
    return response.data;
  },

  process: async (text: string) => {
    const response = await api.post('/intents/process', { text });
    return response.data;
  },
};
